### PR TITLE
formatter fixv2

### PR DESCRIPTION
**Pull request by jd-lara**
@daniel-thom this should prevent dealing with formatter versions in your local environment. I also added new flags.

### CODE REVIEW

**Comment by daniel-thom** [comment]
The cases where the formatter "fixed" leaky returns makes the code less clear. Shall I go back through this code and add explicit returns?

File: src/utils/lazy_dict_from_iterator.jl [hunk 1]
```diff
@@ -70,12 +70,12 @@ end
 """Reset the iterator for cases where underlying arrays have changed."""
 function reset_iterator(container::LazyDictFromIterator)
     @debug "reset_iterator"
-    container.state = nothing
+    return container.state = nothing
```
Comment (jd-lara) [inline src/utils/lazy_dict_from_iterator.jl:73]: These ones need manual fixing

File: .github/workflows/formatter/formatter_code.jl [hunk 2]
```diff
@@ -11,7 +17,8 @@ for main_path in main_paths
             remove_extra_newlines = true,
             verbose = true,
             always_for_in = true,
-            whitespace_typedefs = true
+            whitespace_typedefs = true,
+            always_use_return = true
```
Comment (daniel-thom) [inline .github/workflows/formatter/formatter_code.jl:21]: Just to confirm: this needs to be set to false, right?

Comment (jd-lara) [reply .github/workflows/formatter/formatter_code.jl:21]: correct

**Pull request merged**